### PR TITLE
Fix error when logging segmentation image

### LINKED ISSUE #6443: Unexpected error when logging Segmentation Image and Annotation Context together
### Repro

This fails:
```py
import numpy as np
import rerun as rr

image = np.zeros((8, 12), dtype=np.uint8)
image[0:4, 0:6] = 1
image[4:8, 6:12] = 2

rr.init("rerun_example_segmentation_image", spawn=True)

rr.log(
    "/image",
    rr.SegmentationImage(image),
    rr.AnnotationContext([(1, "red", (255, 0, 0)), (2, "green", (0, 255, 0))]),
)
```

And produces the error:
```
[2024-05-27T21:27:54Z ERROR re_space_view_spatial::visualizers::images] Failed to create rectangle draw data from images: Color mapping is being applied to a four-component RGBA texture
```

This works:
```py
import numpy as np
import rerun as rr

image = np.zeros((8, 12), dtype=np.uint8)
image[0:4, 0:6] = 1
image[4:8, 6:12] = 2

rr.init("rerun_example_segmentation_image", spawn=True)

rr.log(
    "/image",
    rr.SegmentationImage(image),
)

rr.log(
    "/image",
    rr.AnnotationContext([(1, "red", (255, 0, 0)), (2, "green", (0, 255, 0))]),
)

```

### Expected Behavior
These should be equivalent.

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #6443
    * the problem was that we use row ids for hashing texture manager entries. Instead, we need to use a combination of texture meaning + row id, because the same row may give rise to several textures! 
* Fixes this also for any other combination of textures on the same row (e.g. depth + regular)
* Fixes bad error message display

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6449?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6449?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6449)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_viewer_context/src/gpu_bridge/tensor_to_gpu.rs [hunk 1]
```diff
@@ -27,6 +27,27 @@ use super::{get_or_create_texture, try_get_or_create_texture};
...
+    TensorData(TensorDataMeaning),
+}
+
+/// Returns a texture key for a given row id & usage.
+///
+/// Several textures may be created from the same row.
+/// This makes sure that they all get different keys!
+fn generate_texture_key(row_id: RowId, usage: TextureKeyUsage) -> u64 {
+    hash(row_id)
+        ^ hash(match usage {
```
[the remark this answers is not in the document]
Comment (Wumpf) [reply crates/re_viewer_context/src/gpu_bridge/tensor_to_gpu.rs]: 🤦 ofc. Thanks!

**Review by abey79 — APPROVED** [review]
I didn't get all the subtleties of the original issue, but tested with Jeremy's repro and code lgtm

File: crates/re_viewer_context/src/gpu_bridge/tensor_to_gpu.rs [hunk 2]
```diff
@@ -27,6 +27,30 @@ use super::{get_or_create_texture, try_get_or_create_texture};
...
+        row_id,
+        match usage {
+            TextureKeyUsage::TensorData(meaning) => match meaning {
+                TensorDataMeaning::Unknown => 0x12345678,
+                TensorDataMeaning::ClassId => 0x23456789,
+                TensorDataMeaning::Depth => 0x34567890,
+            },
+            TextureKeyUsage::AnnotationContextColormap => 0x45678901,
+        },
+    ))
```
Comment (emilk) [inline crates/re_viewer_context/src/gpu_bridge/tensor_to_gpu.rs:51]:
```suggestion
    hash((row_id, meaning))
```

File: crates/re_viewer_context/src/gpu_bridge/tensor_to_gpu.rs [hunk 3]
```diff
@@ -27,6 +27,30 @@ use super::{get_or_create_texture, try_get_or_create_texture};
 
 // ----------------------------------------------------------------------------
 
+#[derive(Copy, Clone)]
```
Comment (emilk) [inline crates/re_viewer_context/src/gpu_bridge/tensor_to_gpu.rs:30]:
```suggestion
#[derive(Copy, Clone, Hash)]
```

Comment (Wumpf) [reply crates/re_viewer_context/src/gpu_bridge/tensor_to_gpu.rs:30]: oh. wow. of course. I started out with fixing a single value and then ended up creating this enum without thinking to hash it

**Pull request merged**